code/segmentedtickerplant/datastripe.q

```diff
@@ -48,4 +48,32 @@ segmentedsubdetails: {[tabs;instruments;id] (!). flip 2 cut (
         `filters ; .stpps.filtermap[tabs;id]                                            //
         )}
 
+\d .stpps
+//function to grab where clause from config tables
+segmentfilter:{[tbl;segid]
+  wcref:first exec wcRef from .stpps.segmentconfig where table=tbl , segmentID=segid;
+  if[wcref~`;
+    .lg.o["Invalid pairing of table ",string[tbl]," and segmentID ",string[segid],""]];
+  .stpps.segmentfiltermap[wcref]
+  };
+
+  // Function to subscribe to particular segment using segmentID based on .u.sub
+subsegment:{[tbl;segid];
+//tablename and segmentid used to get filters
+  if[tbl~`;:.z.s[;segid] each .stpps.t];
+   if[not tbl in .stpps.t;
+     .lg.e[`sub;m:"Table ",string[tbl]," not in list of stp pub/sub tables"];
+     :(tbl;m)
+  ];
+  filter:segmentfilter[tbl;segid];
+  if[filter~"";
+    .lg.o[`sub;m:"Incorrect pairing of table ",string[tbl]," and segmentID ",string[segid]," not found in .stpps.segmentconfig"];
```

Comment:
can we change this to .lg.e, to match original .u.sub functionality. .proc.stop=0b on stp so wont kill process.